### DISCUSSION TITLE
Is it possible to reference method's parameters inside v_if sentence?

### DISCUSSION THREAD

**Discussion by jealtw** [Kitware/trame#554] [Q&A] [answered]
I have the following code:

```python
def common_params_table(**category**):

    title = "TITLE A" if category == 1 else "TITLE B" if category == 2 else "TITLE C" 
    with param_card(title):
        with html.Table():
            with html.Thead():
                with html.Tr():
                    html.Th("", classes="text-left", width=40)
                    html.Th("Column1 Title", classes="text-left", width=120)
                    html.Th("Column2 Title", classes="text-left", width=110)
                    html.Th("Column3 Title", classes="text-left", width=110)
                    html.Th("Column4 Title", classes="text-left", width=120)
            with html.Tbody(v_for="(item, index) in Params", key="index", value=["item"]):
                with html.Tr(v_if="item['category']==category"):
                    with html.Td(v_if="item['displayable']", classes="text-left"):
                        ......
```

I want to reference '**category**' parameter of  '**common_params_table**' method inside v_if sentence like following:
 with html.Tr(**v_if="item['category']==category"**):

Is it possible?
And if it is possible, what is the correct syntax? Please give any help. Thanks in advance.

**Comment by jourdain** [comment]
I don't understand what you are trying to do. But if you want to leverage some JS logic, you need to create the variables you need onto your server.state (i.e. sever.state.category = 1). And then you can do whatever you want.

**Reply by jealtw** [reply]
So the conclusion is : if 'category' is not created on server.state, it is just a parameter of method common_params_table(category), we can not use v_if="item['category']==category" or some code similar.  Is that right?

**Reply by jealtw** [reply]
I just trying to filter data by category. For example, in the above code, Params is the whole model data defined on server.state, I want to use common_params_table(category) method to display different category data in different table. The UI looks like the following:

Panel
     Table-1 for Category1
     Table-2 for Category2
     Table-3 for Category3

One solution to realize the above requirement I know is to split Params to 3 parts,  that is to define 3 different params on server.state (server.state.ParamsCategory1, server.state.ParamsCategory2, server.state.ParamsCategory3,), and bind them on each table. But that will lead to some repeat code. I wonder if I can realize it with only one method without repeat code.

**Comment by jourdain** [comment, marked as answer]
```python
def common_params_table(category):
    with html.Tbody(v_for="(item, index) in Params", key="index"):
        with html.Tr(v_if=f"item.category == {category}"):
             ...
```

**Reply by jealtw** [reply]
Thanks for your help.
